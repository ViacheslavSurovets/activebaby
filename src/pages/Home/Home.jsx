import React from 'react';
import { SliderMain, CardContainer, SliderSmall, VisitUsContainer } from '@components';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import {
  selectSliderHugeFirstData,
  selectSliderHugeSecondData,
  selectSliderSmallFirstData,
  selectSliderSmallSecondData
} from '@redux/homeData/homeData.selectors';
import PropTypes from 'prop-types';

const Home = ( props ) => {
  const {
    sliderHugeFirstData,
    sliderHugeSecondData,
    sliderSmallFirstData,
    sliderSmallSecondData
  } = props;
  return (
    <>
      <SliderMain items={ sliderHugeFirstData () } />
      <CardContainer />
      <SliderSmall items={ sliderSmallFirstData () } />
      <SliderMain items={ sliderHugeSecondData () } />
      <SliderSmall items={ sliderSmallSecondData () } />
      <VisitUsContainer />
    </>
  );
};

const mapStateToProps = createStructuredSelector ( {
  sliderHugeFirstData: selectSliderHugeFirstData,
  sliderHugeSecondData: selectSliderHugeSecondData,
  sliderSmallFirstData: selectSliderSmallFirstData,
  sliderSmallSecondData: selectSliderSmallSecondData
} );

export default connect ( mapStateToProps ) ( Home );


Home.propTypes = {
  sliderHugeFirstData: PropTypes.func,
  sliderHugeSecondData: PropTypes.func,
  sliderSmallFirstData: PropTypes.func,
  sliderSmallSecondData: PropTypes.func
};




