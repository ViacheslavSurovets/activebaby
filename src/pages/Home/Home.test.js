import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import Home from './Home';
import configureMockStore  from 'redux-mock-store';


const mockStore = configureMockStore();
const store = mockStore({});

describe('Home', ()=> {
  it('Home Snapshot', ()=> {
    expect ( shallow (
      <Provider store={store}>
        <Home  />
      </Provider>
    ) ).toMatchSnapshot();
  });
});
