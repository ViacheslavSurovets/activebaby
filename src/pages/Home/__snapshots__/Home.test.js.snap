// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Home Snapshot 1`] = `ShallowWrapper {}`;
